src/gulpfile.js

```diff
@@ -186,74 +186,56 @@ gulp.task('chat-styles', ['chat-styles-handlebars', 'chat-styles-js'], function(
 
 // Compile and concatenate handlebars files for all chat styles.
 gulp.task('chat-styles-handlebars', function() {
-    var promises = [];
-    fs.readdir(config.chatStylesPath, function(err, list){
-        if(err) return done(err);
-        list.filter(function(path){
-            return fs.lstatSync(config.chatStylesPath + "/" + path).isDirectory();
-        }).map(function(dir){
-            promises.push(new Promise(function(resolve, reject){
-                var pipeline = gulp.src(config.chatStylesPath + '/' + dir + '/templates_src/client_side/**/*.handlebars')
-                  .pipe(handlebars({
-                      // Use specific version of Handlebars.js
-                      handlebars: handlebarsEngine
-                  }))
-                  .pipe(wrapHandlebarsTemplate())
-                  .pipe(concat('templates.js'))
-                  .pipe(uglify({preserveComments: 'some'}))
-                  .pipe(header(config.compiledTemplatesHeader))
-                  .pipe(gulp.dest(config.chatStylesPath + '/' + dir + '/templates_compiled/client_side'));
-                pipeline.on('end', resolve);
-                pipeline.on('error', reject);
-            }));
+    return Promise.all(getChildDirs(config.chatStylesPath).map(function (dir) {
+        return new Promise(function(resolve, reject){
+            gulp.src(config.chatStylesPath + '/' + dir + '/templates_src/client_side/**/*.handlebars')
+                .pipe(handlebars({
+                    // Use specific version of Handlebars.js
+                    handlebars: handlebarsEngine
+                }))
+                .pipe(wrapHandlebarsTemplate())
+                .pipe(concat('templates.js'))
+                .pipe(uglify({preserveComments: 'some'}))
+                .pipe(header(config.compiledTemplatesHeader))
+                .pipe(gulp.dest(config.chatStylesPath + '/' + dir + '/templates_compiled/client_side'))
+                .on('end', resolve)
```

Comment:
If this code was in a `then` function I don't think the `resolve` function will exist, but it could simply return a value.

Or, like the MDN docs show, if we want to still use the end and error stream events this code we can just return a promise inside the then function. That might actually work out the best.